### ISSUE TITLE
Rework SCA Policy for SUSE Linux Enterprise 15

### ISSUE THREAD

**Issue by 72nomada** [wazuh/wazuh#15167]
| Component | Action type           | Main Issue 
| --- | --- | ---
| SCA       | Rework |  #18306

## Main tasks

- [x] Use the latest CIS benchmark PDF from https://downloads.cisecurity.org/#/ -  CIS SUSE Linux Enterprise 15 Benchmark v1.1.1
- [x] Verify IDs numbers.
- [x] Verify texts are correct: Title, Description, Rationale and Remediation.
- [x] Verify Compliance: CIS, CIS_CSC.
- [x] Verify condtion and rules:
  - [x] To Pass.
  - [x] To Fail.

### Syntax and semantic

- [x] a) ID of each policy must be contiguous.
- [x] b) The order and format set in [Documentation](https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/creating-custom-policies.html) must be respected.
- [x] c) YML must be valid to avoid errors.

### Content

- [x] a) Compare each check with its analogue from CIS Benchmark.
- [x] b) Try to maintain each rule as similar as possible with the `Audit` section from the CIS check.
- [x] c) Check that the commands provide the expected output.
- [x] d) When a failure is discovered, check similar policies to avoid repetition of the issue.

### Unit testing

- [x] a) Output from `ossec.log` after the SCA scan and a raw output of the result of the checks.
<details><summary>Tests results</summary>
<p>

#### Analysisd (server or local)
analysisd.debug=2

#### Auth daemon debug (server)
authd.debug=0

#### Exec daemon debug (server, local, or Unix agent)
execd.debug=0

#### Monitor daemon debug (server, local, or Unix agent)
monitord.debug=0

#### Log collector (server, local or Unix agent)
logcollector.debug=0

#### Integrator daemon debug (server, local or Unix agent)
integrator.debug=0

#### Unix agentd
agent.debug=2

```
2023/11/03 20:33:53 sca[22891] wm_sca.c:2598 at wm_sca_send_summary(): DEBUG: Sending summary event for file: 'cis_sles15_linux.yml'
2023/11/03 20:33:53 sca[22891] wm_sca.c:271 at wm_sca_send_alert(): DEBUG: Sending event: {"type":"summary","scan_id":2057074577,"name":"CIS SUSE Linux Enterprise 15 Benchmark v1.1.1","policy_id":"cis_sles15_linux","file":"cis_sles15_linux.yml","description":"This document provides prescriptive guidance for establishing a secure configuration posture for SUSE Linux Enterprise 15 systems running on x86 and x64 platforms. This document was tested against SUSE Linux Enterprise Server 15 SP1.","references":"https://www.cisecurity.org/cis-benchmarks/","passed":73,"failed":84,"invalid":12,"total_checks":169,"score":46.4968147277832,"start_time":1699043626,"end_time":1699043630,"hash":"9f193a571f070d6194d47afa9b563e11dac8897a1d218c7843bb282d6a71ac2b","hash_file":"f602b87ec68463fd7546242abe5fdee6d9d8f8cf13aad98a7270f806eeed05d6","first_scan":1}
```
</p>
</details>

### Deployment

- [x] a) If the policy it's new, it must be added to the `sca.files` templates.
- [x] b) If the OS has many supported SCA policies, a policy must be set as default policy. ([as example](https://github.com/wazuh/wazuh/blob/master/etc/templates/config/centos/sca.files))

**Comment by IsExec** [comment]
## Update 12/09/2023
https://github.com/wazuh/wazuh/pull/18985/commits/ff34c2ac6b3e87bb1de61a22ab73a5acd4cf0ae3
SUSE Linux Enterprise 15 SCA rework IDs fixed
https://github.com/wazuh/wazuh/pull/18985/commits/deb04a206fdd4c6045ab2ae49a8ebc4c2db034e1
SUSE Linux Enterprise SCA rework section 1.1.1.1-1.1.23

**Comment by jk-olaoluwa** [comment]
### Update 30/10/2023
#### Preparing test environment for SUSE Linux Enterprise 15

**Comment by IsExec** [comment]
## Review correction - 31/10/2023

https://github.com/wazuh/wazuh/issues/15167#issuecomment-1784750533
https://github.com/wazuh/wazuh/pull/18985/commits/4f482835bbba6d39d72cde1f89b6abd3f1f26eb0

1.1.3 - Remain unchanged. The PDF uses grep -v "string" which takes input line by line, and outputs only the lines in which "string" does not appear. The best way to represent this is o have not and !r. That way only lines with 'tmp' and 'noexec' will pass.
1.1.4 - Remain unchanged. Same reason as 1.1.3
1.1.5 - Remain unchanged. Same reason as 1.1.3
1.1.7 - Remain unchanged. Same reason as 1.1.3
1.1.8 - Remain unchanged. Same reason as 1.1.3
1.1.9 - Remain unchanged. Same reason as 1.1.3
1.1.12 - Remain unchanged. Same reason as 1.1.3
1.1.13 - Remain unchanged. Same reason as 1.1.3
1.1.14 - Remain unchanged. Same reason as 1.1.3
1.1.18 - Remain unchanged. Same reason as 1.1.3
1.6.1 -  Fixed as not implemented due SCA capabilities limitation of A and (B or C) logic
1.7.1.1 - Fixed
1.8.1.3 - Fixed
1.9 - Fixed
1.10 - Fixed

**Comment by ooniagbi** [comment]
Reviewed and tested successfully.

**Comment by IsExec** [comment]
Excluded rules with grep and removed manual check 1.9 to avoid high CPU Utilization and extended time of completion - https://github.com/wazuh/wazuh/pull/24975/commits/7d884b278f9be3fc9cbfb4372cf429e3034a5453

**Comment by IsExec** [comment]
# SUSE Linux Enterprise 15 SCA Testing Report
**Condition**: The reworked SCA was benchmarked against old SUSE Linux Enterprise 15 SCA.  

## Agent running on old SCA (SUSE Linux Enterprise 15 SCA)

###  Endpoint spec

<details>
<summary>CPU</summary>

````console
# lscpu
Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          39 bits physical, 48 bits virtual
  Byte Order:             Little Endian
CPU(s):                   2
  On-line CPU(s) list:    0,1
Vendor ID:                GenuineIntel
  Model name:             Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    BIOS Model name:        CPU @ 0.0GHz
    BIOS CPU family:      0
    CPU family:           6
    Model:                142
    Thread(s) per core:   1
    Core(s) per socket:   2
    Socket(s):            1
    Stepping:             10
    BogoMIPS:             4223.99
    Flags:                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm
                          constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 movbe po
                          pcnt aes rdrand hypervisor lahf_lm abm 3dnowprefetch pti fsgsbase bmi1 bmi2 invpcid rdseed clflushopt md_clear flush_l1d a
                          rch_capabilities
Virtualization features:
  Hypervisor vendor:      KVM
  Virtualization type:    full
Caches (sum of all):
  L1d:                    64 KiB (2 instances)
  L1i:                    64 KiB (2 instances)
  L2:                     512 KiB (2 instances)
  L3:                     16 MiB (2 instances)
NUMA:
  NUMA node(s):           1
  NUMA node0 CPU(s):      0,1
````

</details>
<details>
<summary>RAM</summary>

````console
# free -g -h -t
               total        used        free      shared  buff/cache   available
Mem:           3.8Gi       391Mi       2.6Gi        16Mi       1.0Gi       3.4Gi
Swap:          511Mi          0B       511Mi
Total:         4.3Gi       391Mi       3.1Gi
````

</details>

<details>
<summary>Disk</summary>

````console
# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda3        64G  2.3G   58G   4% /
devtmpfs        4.0M     0  4.0M   0% /dev
tmpfs           1.9G     0  1.9G   0% /dev/shm
tmpfs           770M   17M  754M   3% /run
/dev/sda3        64G  2.3G   58G   4% /opt
/dev/sda3        64G  2.3G   58G   4% /boot/grub2/i386-pc
/dev/sda3        64G  2.3G   58G   4% /srv
/dev/sda3        64G  2.3G   58G   4% /usr/local
/dev/sda3        64G  2.3G   58G   4% /boot/grub2/x86_64-efi
/dev/sda3        64G  2.3G   58G   4% /home
/dev/sda3        64G  2.3G   58G   4% /tmp
/dev/sda3        64G  2.3G   58G   4% /var/crash
/dev/sda3        64G  2.3G   58G   4% /var/lib/mailman
/dev/sda3        64G  2.3G   58G   4% /var/lib/named
/dev/sda3        64G  2.3G   58G   4% /var/opt
/dev/sda3        64G  2.3G   58G   4% /var/tmp
/dev/sda3        64G  2.3G   58G   4% /var/log
/dev/sda3        64G  2.3G   58G   4% /var/lib/pgsql
/dev/sda3        64G  2.3G   58G   4% /var/spool
vagrant         953G  498G  456G  53% /vagrant
tmpfs           385M  4.0K  385M   1% /run/user/1000
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/348c6003-d62f-44ab-9495-44c34ad4bfb2)
![disk_percent_plot](https://github.com/user-attachments/assets/a12f5deb-e0c2-4578-b7e5-cbc2e27c8102)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/668683c6-c94c-4ee8-bb72-80e1781a90a5)
![memory_percent_plot](https://github.com/user-attachments/assets/bfe605f5-8852-488a-ac39-96165e2154ee)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/307cfac4-a0ee-465c-8803-86d1e62de37b)
![ProcessesCPU](https://github.com/user-attachments/assets/ffcb4763-c49b-47fc-84e5-408dc0ad13ba)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/99dc8391-b3e7-44c9-86f6-0a6e7d68675f)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/c251dc70-66ac-4847-9dea-11d44237cdc1)
![ProcessesRAM](https://github.com/user-attachments/assets/b015e6b1-8dd3-49fa-8560-8ee90c166a8f)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/1e4af084-7c5e-44eb-88bf-56c7dda11314)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/5404d9a0-c56e-49e7-8f13-3382d23a9b57)
![swap_percent_plot](https://github.com/user-attachments/assets/fb41f56e-6c83-468c-a130-7f1a2aeb46db)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/7b004e0a-29d0-49a8-aa54-0b565ca27456)
</details>

<details>
<summary>SCA logs</summary>

````console
# tail /var/ossec/logs/ossec.log
2024/07/30 23:13:56 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/30 23:13:56 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/30 23:14:03 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/30 23:14:03 sca: INFO: Security Configuration Assessment scan finished. Duration: 7 seconds.
2024/07/30 23:18:56 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/30 23:18:56 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/30 23:19:03 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/30 23:19:03 sca: INFO: Security Configuration Assessment scan finished. Duration: 7 seconds.
2024/07/30 23:23:56 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/30 23:23:56 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/30 23:24:04 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/30 23:24:04 sca: INFO: Security Configuration Assessment scan finished. Duration: 8 seconds.
2024/07/30 23:28:56 sca: INFO: Starting Security Configuration Assessment scan.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16436605/metrics.zip)


## Agent running on reworked SCA (SUSE Linux Enterprise 15)

### Endpoint spec
<details>
<summary>CPU</summary>

````console
# lscpu
Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          39 bits physical, 48 bits virtual
  Byte Order:             Little Endian
CPU(s):                   2
  On-line CPU(s) list:    0,1
Vendor ID:                GenuineIntel
  Model name:             Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    BIOS Model name:        CPU @ 0.0GHz
    BIOS CPU family:      0
    CPU family:           6
    Model:                142
    Thread(s) per core:   1
    Core(s) per socket:   2
    Socket(s):            1
    Stepping:             10
    BogoMIPS:             4223.99
    Flags:                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm
                          constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 movbe po
                          pcnt aes rdrand hypervisor lahf_lm abm 3dnowprefetch pti fsgsbase bmi1 bmi2 invpcid rdseed clflushopt md_clear flush_l1d a
                          rch_capabilities
Virtualization features:
  Hypervisor vendor:      KVM
  Virtualization type:    full
Caches (sum of all):
  L1d:                    64 KiB (2 instances)
  L1i:                    64 KiB (2 instances)
  L2:                     512 KiB (2 instances)
  L3:                     16 MiB (2 instances)
NUMA:
  NUMA node(s):           1
  NUMA node0 CPU(s):      0,1
````

</details>

<details>
<summary>RAM</summary>

````console
# free -g -h -t
               total        used        free      shared  buff/cache   available
Mem:           3.8Gi       415Mi       2.6Gi        16Mi       1.0Gi       3.4Gi
Swap:          511Mi          0B       511Mi
Total:         4.3Gi       415Mi       3.1Gi
````

</details>

<details>
<summary>Disk</summary>

````console
# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda3        64G  2.3G   58G   4% /
devtmpfs        4.0M     0  4.0M   0% /dev
tmpfs           1.9G     0  1.9G   0% /dev/shm
tmpfs           770M   17M  754M   3% /run
/dev/sda3        64G  2.3G   58G   4% /boot/grub2/i386-pc
/dev/sda3        64G  2.3G   58G   4% /boot/grub2/x86_64-efi
/dev/sda3        64G  2.3G   58G   4% /var/spool
/dev/sda3        64G  2.3G   58G   4% /srv
/dev/sda3        64G  2.3G   58G   4% /tmp
/dev/sda3        64G  2.3G   58G   4% /home
/dev/sda3        64G  2.3G   58G   4% /opt
/dev/sda3        64G  2.3G   58G   4% /var/tmp
/dev/sda3        64G  2.3G   58G   4% /var/opt
/dev/sda3        64G  2.3G   58G   4% /var/log
/dev/sda3        64G  2.3G   58G   4% /var/lib/pgsql
/dev/sda3        64G  2.3G   58G   4% /var/lib/named
/dev/sda3        64G  2.3G   58G   4% /var/lib/mailman
/dev/sda3        64G  2.3G   58G   4% /var/crash
/dev/sda3        64G  2.3G   58G   4% /usr/local
vagrant         953G  499G  454G  53% /vagrant
tmpfs           385M  4.0K  385M   1% /run/user/1000
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/ffd4d98d-48bf-4aa4-8802-863b97c513e5)
![disk_percent_plot](https://github.com/user-attachments/assets/dc8f4b63-f310-4f45-95ad-4f4014d7e9fb)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/fe8e8166-fe9e-4abf-a614-87a205abe273)
![memory_percent_plot](https://github.com/user-attachments/assets/570fbf25-8a05-4f92-bbb5-122f9ceaa68d)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/6cdff00a-f4ff-421c-aeef-930e38495959)
![ProcessesCPU](https://github.com/user-attachments/assets/d502c48f-6ed3-48e7-9ef2-fb119bb3c02e)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/7243fe1e-9707-4477-8474-7a594351bbc1)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/b156fe02-a5df-4823-9e52-17aaf25b776c)
![ProcessesRAM](https://github.com/user-attachments/assets/540d4d0e-095a-49be-960f-6b18d3dfe46f)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/53102e86-6efa-4e9c-85ba-07d329fc0dbb)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/61423bc3-956b-4478-bb70-75ae1fae6b48)
![swap_percent_plot](https://github.com/user-attachments/assets/fcaa14e6-cfe7-45cd-b738-5623afdb76fb)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/aaaf357e-5d76-4a6c-a25e-7de59aff32a1)
</details>

<details>
<summary>SCA logs</summary>

````console
# tail /var/ossec/logs/ossec.log
2024/07/31 04:44:33 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/31 04:44:33 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
2024/07/31 04:49:29 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 04:49:29 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/31 04:49:33 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/31 04:49:33 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
2024/07/31 04:54:29 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 04:54:29 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/31 04:54:34 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_sles15_linux.yml'
2024/07/31 04:54:34 sca: INFO: Security Configuration Assessment scan finished. Duration: 5 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16436629/metrics.zip)


## Analysis

  | Benchmark (SLES   15 Old) | New SCA (SLES   15 rework) | Difference
-- | -- | -- | --
CPU |   |   |  
`wazuh-modulesd` average   CPU percent | 3.13 | 1.4 | -1.73
Overall   average CPU percent | 14.63 | 5.4 | -9.23
Overall peak   CPU percent | 22.5 | 8.3 | -14.2
Memory |   |   |  
`wazuh   modulesd` average memory (MB) | 0.02 | 0.02 | 0
Overall   average memory used (MB) | 175.96 | 175.54 | -0.42
Overall peak   memory used peak (MB) | 184.86 | 182.04 | -2.82
Disk |   |   |  
Overall   average disk used (MB) | 2288.46 | 2164.91 | -123.55
Overall peak   disk used (MB) | 2288.46 | 2164.91 | -123.55
Overall   average disk read bytes | 1990.6 | 2004.5 | 13.9
Overall peak   disk read bytes | 1990.6 | 2004.5 | 13.9
Overall   average disk write bytes | 1165.38 | 1022.43 | -142.95
Overall peak   disk write bytes | 1165.8 | 1022.43 | -143.37

**Comment by ooniagbi** [comment]
LGTM!

**Comment by ooniagbi** [comment]
Merged to 4.10.0
